tests/go_test.go

```diff
@@ -1375,6 +1377,16 @@ func CheckStringIsNestedError(fail func(string, ...interface{})) {
 	b.CreateString("foo")
 }
 
+func CheckSharedStrings(fail func(string, ...interface{})) {
+	b := flatbuffers.NewBuilder(0)
+	add := func(s string) flatbuffers.UOffsetT {
+		return b.CreateSharedString(s)
+	}
+	if err := quick.CheckEqual(add, add, nil); err != nil{
+		fail("expected same offset")
+	}
```

Comment:
I'd like another quickcheck (that might supersede this one): the input is a slice of strings, and every pairwise combination of those strings is checked against your `CreateSharedString` function. If the strings are equal, the offsets should be equal, and if the strings are unequal, they offsets should be unequal. This tests both the positive and negative logic.

Pseudocode scaffold:

```golang
f := func(xs: []string) bool {
   bldr := flatbuffers.Newbuilder(0)

   for a in xs {
      for b in xs {
         offA = ...
         offB = ...

         if (a == b) && (offA != offB) { return false }
         if (a != b) && (offA == offB) { return false }
      }
    }

    return true
}

if err := quick.Check(f, nil); ....
```

Reply:
should be alright now